django/db/models/deletion.py

```diff
@@ -31,6 +39,12 @@ def PROTECT(collector, field, sub_objs, using):
     )
 
 
+def RESTRICT(collector, field, sub_objs, using):
+    collector.add_restricted_objects(field, sub_objs)
+    if not field.null:
+        collector.add_dependency(field.remote_field.model, field.model)
```

Comment:
Do we need this change? If yes then tests are missing. IMO it is not necessary.

Reply:
Yes, tests will fail without this change when using MySQL